mlibrary/lib.umich.edu
# Website 39 add onsite location details to where for events exhibits page template

# Overview
This PR is specific to the "WHERE" for event details on [the event template ](https://lib.umich.edu/visit-and-study/events-and-exhibits/today-and-upcoming/silenced-muse-deciphering-emily-hales-legacy) and the event cards as seen on the [Today and Upcoming page](https://lib.umich.edu/visit-and-study/events-and-exhibits/today-and-upcoming) and the "What’s happening?" section on [the homepage](https://lib.umich.edu/).

There are three parts to this enhancement:
- Changing how we present the WHERE for events/exhibits in library locations to include the floor, room # (where relevant), and “view floorplan” link, modeled after the Designation page templates
- Adding a “View directions” link to the WHERE for events/exhibits non-library locations
- Adjusting the presentation for Hybrid events to align with the other changes

> This pull request addresses [WEBSITE-39](https://mlit.atlassian.net/browse/WEBSITE-39?focusedCommentId=363458&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-363458).

## How I went about it
The strings are built in `eventFormatWhere` in the event.js utility file. They're built by pushing a series of strings named `label` to an object array `where`. `where` is then mapped in the event template file and the `locality` is extracted and applied as a link if it's present.

The floor plans are loaded in the event template file using `useFloorPlan` from use-floor-plan.js.

The "View directions" link is generated from `createGoogleMapsURL` if there is a locality in there `where` object. Both non-library events and library events can have a "View directions" link, but non-library events should **always** have a "View directions" link.

## Examples
Here are some current events that we can preview:
- A[ non-library location](https://deploy-preview-313--future-wwwlib-previews.netlify.app/visit-and-study/events-and-exhibits/today-and-upcoming/silenced-muse-deciphering-emily-hales-legacy) with a “View Directions” link.
- A [library location ](https://deploy-preview-313--future-wwwlib-previews.netlify.app/visit-and-study/events-and-exhibits/today-and-upcoming/international-studies-horror-filmfest)with a “View floorplan” link.
- A [library location](https://deploy-preview-313--future-wwwlib-previews.netlify.app/visit-and-study/events-and-exhibits/today-and-upcoming/clements-treasures-items-and-about-mixed-race-people) with an address and “View Directions” link.

For other events, hybrid and online, we don’t have current examples, but I can edit event details before running the logic to see what that looks like:

- A hybrid event with a library location:
![image](https://github.com/user-attachments/assets/8c45811c-593a-4642-bcc7-18071d00e439)
And the card:
![image](https://github.com/user-attachments/assets/0c8ea818-d6c9-4b7e-9a43-298c87e19f77)

- A hybrid event with a library location and event link:
![image](https://github.com/user-attachments/assets/f615115c-7bba-4346-8e20-0c88651ba4fb)
And the card (same as above):
![image](https://github.com/user-attachments/assets/f1ae8e30-3998-4265-b40e-dc7ad1b7fe4c)

- An online event (no library location):
![image](https://github.com/user-attachments/assets/c10b087a-095d-428b-a2f7-566cc100baf0)
And the card:
![image](https://github.com/user-attachments/assets/dde3fba4-744b-4b07-8cc1-7564d26053cf)

## Testing
- Make sure the PR is consistent in these browsers:
- Run accessibility tests:
